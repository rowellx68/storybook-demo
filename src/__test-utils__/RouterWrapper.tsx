import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

const RouterWrapper: React.FunctionComponent = ({ children }) => {
  return (
    <MemoryRouter keyLength={0}>
      <Route exact path='/' />
      {children}
    </MemoryRouter>
  );
};

export default RouterWrapper;
