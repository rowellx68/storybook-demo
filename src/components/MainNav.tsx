/** @jsx jsx */
import React, { ReactNode, useContext } from 'react';
import { css, jsx } from '@emotion/core';

import { NavLink } from 'react-router-dom';
import { shadows, colours } from 'styles/variables';
import PageContext from 'contexts/PageContext';

const nav = css(shadows.default, {
  backgroundColor: colours.primary,
  height: 50,
  zIndex: 1,
  transition: 'background-color 200ms ease-in',
});

const navItem = css({
  color: '#ffffff',
  height: 50,
  lineHeight: '50px',
  display: 'inline-block',
  textDecoration: 'none',
});

const navItemBrand = css(navItem, {
  fontWeight: 'bold',
  padding: '0 1.25rem',
});

interface Props {
  children?: ReactNode;
}

const MainNav: React.FunctionComponent<Props> = ({ children }) => {
  const { name } = useContext(PageContext);
  const isAdminPage = name === 'admin';

  return (
    <nav css={nav}>
      <NavLink exact to='/' css={navItemBrand}>
        {isAdminPage ? 'HP | Admin' : 'Holiday Planner'}
      </NavLink>
    </nav>
  );
};

export default MainNav;
