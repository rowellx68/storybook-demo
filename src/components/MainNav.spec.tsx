import React from 'react';
import { mount } from 'enzyme';

import RouterWrapper from '__test-utils__/RouterWrapper';
import MainNav from 'components/MainNav';

const options = {
  wrappingComponent: RouterWrapper
};

describe('MainNav', () => {
  it('renders the nav', () => {
    const component = mount(<MainNav />, options);

    expect(component.exists('nav')).toEqual(true);
  });

  it('renders the correct mark-up', () => {
    const component = mount(<MainNav />, options);

    expect(component).toMatchSnapshot();
  });
});
