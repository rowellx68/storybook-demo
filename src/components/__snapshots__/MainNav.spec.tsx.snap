// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav renders the correct mark-up 1`] = `
.emotion-4 {
  box-shadow: 0 1px 3px 0 rgba(0,0,0,0.1),0 1px 2px 0 rgba(0,0,0,0.06);
  background-color: var(--primary);
  height: 50px;
  z-index: 1;
  -webkit-transition: background-color 200ms ease-in;
  transition: background-color 200ms ease-in;
}

.emotion-0 {
  color: #ffffff;
  height: 50px;
  line-height: 50px;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: bold;
  padding: 0 1.25rem;
}

<MainNav>
  <nav
    className="emotion-4"
  >
    <NavLink
      className="emotion-0"
      exact={true}
      to="/"
    >
      <Link
        aria-current="page"
        className="emotion-0 active"
        style={Object {}}
        to={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": null,
          }
        }
      >
        <LinkAnchor
          aria-current="page"
          className="emotion-0 active"
          href="/"
          navigate={[Function]}
          style={Object {}}
        >
          <a
            aria-current="page"
            className="emotion-0 active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Holiday Planner
          </a>
        </LinkAnchor>
      </Link>
    </NavLink>
  </nav>
</MainNav>
`;
